martin/src/geojson/mod.rs

```diff
@@ -0,0 +1,117 @@
+use std::fmt::{Debug, Formatter};
+use std::io;
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use geojson_vt_rs::{GeoJSONVT, Options};
+use martin_tile_utils::{Format, TileCoord, TileInfo};
+use prost::Message as _;
+use std::fs::File;
+use tilejson::TileJSON;
+use tilejson::tilejson;
+use tokio::sync::Mutex;
+
+use crate::file_config::FileError::{self, InvalidMetadata, IoError};
+use crate::file_config::FileResult;
+use crate::geojson::mvt::LayerBuilder;
+use crate::source::{TileData, TileInfoSource, UrlQuery};
+use crate::{MartinResult, Source};
+
+mod config;
+mod mvt;
+
+pub use config::GeoJsonConfig;
+
+pub struct GeoJsonSource {
+    id: String,
+    path: PathBuf,
+    inner: Arc<Mutex<GeoJSONVT>>,
+    tilejson: TileJSON,
+    tile_info: TileInfo,
+}
+
+impl Debug for GeoJsonSource {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "GeoJsonSource {{ id: {}, path: {:?} }}",
+            self.id, self.path,
+        )
```

Comment:
nit: I think this is a bit more rusty. Your version is equivalent functionality wise ^^.

```suggestion
        f.debug_struct("GeoJsonSource")
         .field("id", &self.id)
         .field("path", &self.path)
         .finish()
```